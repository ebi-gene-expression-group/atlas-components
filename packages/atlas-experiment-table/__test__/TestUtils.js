import React from 'react'
import styled from 'styled-components'

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

const TableCellDiv = styled.div`
  font-size: 13px;
  font-family: Helvetica, Arial, FreeSans, "Liberation Sans", sans-serif;
`

const tableHeader = [
  { type: `sort`, title: `Loaded date`, width: 140, dataParam: `lastUpdate` },
  { type: `search`, title: `species`, width: 200, dataParam: `species` },
  { type: ``, title: `experiment description`, width: 360, dataParam: `experimentDescription`, link: `experimentAccession`, resource: `experiments`, endpoint: `Results` },
  { type: `plain`, title: `experiment factors`, width: 260, dataParam: `experimentalFactors` },
  { type: `sort`, title: `Number of assays`, width: 160, dataParam: `numberOfAssays`, link: `experimentAccession`, resource: `experiments`, endpoint: `Experiment Design` },
]

const data = [
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-EHCA-2`,
    experimentDescription: `Melanoma infiltration`,
    lastUpdate: `16-11-2018`,
    numberOfAssays: 6638,
    numberOfContrasts: 0,
    species: `Mus musculus`,
    kingdom: `animals`,
    experimentalFactors: [`single cell identifier`, `sampling site`,`time`],
  },
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-GEOD-99058`,
    experimentDescription: `Single cell`,
    lastUpdate: `11-10-2018`,
    numberOfAssays: 250,
    numberOfContrasts: 0,
    species: `Mus musculus`,
    kingdom: `animals`,
    experimentalFactors: [`single cell identifier`]
  },
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-MTAB-5061`,
    experimentDescription: `healthy individuals and type 2 diabetes patients`,
    lastUpdate: `11-10-2018`,
    numberOfAssays: 3514,
    numberOfContrasts: 0,
    species: `Homo sapiens`,
    kingdom: `plants`,
    experimentalFactors: [`single cell identifier`, `disease`],
  }
]

export {getRandomInt, TableCellDiv, tableHeader, data}
