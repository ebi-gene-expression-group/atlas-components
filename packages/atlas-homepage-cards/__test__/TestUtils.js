

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

const dummyCards = () => {
  return [
    {
      iconType: `species`,
      iconSrc: `mus musculus`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
    {
      iconType: `species`,
      iconSrc: `homo sapiens`,
      iconDescription: `Homo sapiens`,
      content: [
        {
          text: `2 experiments`,
        }
      ]
    },
    {
      iconType: `unknown`,
      iconSrc: `http://www.lingyun.com`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
    {
      iconType: `experiments`,
      iconSrc: `http://www.zhao.com`,
      iconDescription: `Homo sapiens`,
      content: [
        {
          text: `2 experiments`,
        }
      ]
    },
    {
      iconType: `image`,
      iconSrc: `logo-goes-here`,
      iconDescription: `An image card`,
      content: [
        {
          text: `Go to experiments`,
        }
      ]
    },
    {
      iconType: `imagespecies`,
      iconSrc: `http://www.lingyun.com`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
  ]
}

export {getRandomInt, dummyCards}