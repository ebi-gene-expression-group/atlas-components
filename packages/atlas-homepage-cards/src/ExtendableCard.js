import React from 'react'
import PropTypes from 'prop-types'

import EbiSpeciesIcon from 'react-ebi-species'

import renderContentListItems from './renderContentListItems'
import modelPropTypes from './modelPropTypes'
import styled from 'styled-components'

const MAX = 5

const CardContentList = styled.ul`
  list-style: none;
  width: 80%;
  margin-left: auto;
  margin-right: auto;
`

class ExtendableCard extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      isHidden: this.props.content && this.props.content.length >= MAX
    }
    this.onClick = this.onClick.bind(this)
  }

  onClick() {
    this.setState({
      isHidden: !this.state.isHidden
    })
  }

  render() {
    const { speciesIconHeight, imageIconHeight, iconType, iconSrc, description, content } = this.props
    const visibleContent = Array.isArray(content) && renderContentListItems(content)

    const cardHero =
      iconType === `species` ?
        <span style={{fontSize: speciesIconHeight}}><EbiSpeciesIcon species={iconSrc} /></span> :
      // iconType === `image` ?  // Margin picked by trial and error to match EbiSpeciesIcon of size 8rem
        <img src={iconSrc} style={{height: imageIconHeight, marginBottom: `2.35rem`, marginTop: `2.50rem`}}/>

    return (
      <div style={{marginBottom:0, paddingBottom: `2rem`, textAlign: `center`}}>
        {
          description && (
            description.url ?
              <h4><a href={description.url}>{description.text}</a></h4> :
              <h4>{description.text}</h4>
          )
        }

        {
          description && (
            description.url ?
              <a href={description.url} style={{borderBottom: 0}}>{cardHero}</a> :
              cardHero
          )
        }

        <CardContentList className={`content`}>
          {
            this.state.isHidden ?
              visibleContent.slice(0, MAX) :
              visibleContent
          }
        </CardContentList>
        {
          Array.isArray(content) && content.length > MAX &&
          <button className={`button`} onClick={this.onClick}>{this.state.isHidden ? `Show all` : `Show fewer`}</button>
        }
      </div>
    )
  }
}

ExtendableCard.propTypes = {
  ...modelPropTypes,
  speciesIconHeight: PropTypes.string,
  imageIconHeight: PropTypes.string
}

ExtendableCard.defaultProps = {
  speciesIconHeight: `8rem`,
  imageIconHeight: `8rem`
}

export default ExtendableCard
