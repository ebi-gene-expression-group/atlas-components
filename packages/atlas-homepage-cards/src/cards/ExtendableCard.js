import React from 'react'
import PropTypes from 'prop-types'

import EbiSpeciesIcon from 'react-ebi-species'

import renderContentListItems from './renderContentListItems'
import cardPropTypes from './cardPropTypes'
import styled from 'styled-components'

const MAX = 5

const CardContentList = styled.ul`
  list-style: none;
  width: 80%;
  margin-left: auto;
  margin-right: auto;
`

class ExtendableSpeciesCard extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      isHidden: this.props.content && this.props.content.length >= MAX
    }
    this.onClick = this.onClick.bind(this)
  }

  onClick() {
    this.setState({
      isHidden: !this.state.isHidden
    })
  }

  render() {
    const { iconType, iconSrc, description, content } = this.props
    const visibleContent = Array.isArray(content) && renderContentListItems(content)

    const cardHero =
      iconType === `species` ?
        <span style={{fontSize: `8rem`}}><EbiSpeciesIcon species={iconSrc} /></span> :
      // iconType === `image` ?  // Margin picked by trial and error to match EbiSpeciesIcon of size 8rem
        <img src={iconSrc} style={{height: `8rem`, marginBottom: `2.35rem`, marginTop: `2.50rem`}}/>

    return (
      <div style={{marginBottom:0, paddingBottom: `2rem`, textAlign: `center`}}>
        {
          description && (
            description.url ?
              <h4><a href={description.url}>{description.text}</a></h4> :
              <h4>{description.text}</h4>
            )
        }

        {
          description && (
            description.url ?
              <a href={description.url} style={{borderBottom: 0}}>{cardHero}</a> :
              cardHero
            )
        }

        <CardContentList className={`content`}>
        {
          this.state.isHidden ?
            visibleContent.slice(0, MAX) :
            visibleContent
        }
        </CardContentList>
        {
          Array.isArray(content) && content.length > MAX &&
          <button className={`button`} onClick={this.onClick}>{this.state.isHidden ? `Show all` : `Show fewer`}</button>
        }
      </div>
    )
  }
}

ExtendableSpeciesCard.propTypes = cardPropTypes

export default ExtendableSpeciesCard
