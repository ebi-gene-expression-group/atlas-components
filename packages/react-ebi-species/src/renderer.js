"use strict";

//*------------------------------------------------------------------*

var React = require('react');
var ReactDOM = require('react-dom');

//*------------------------------------------------------------------*

var Icon = require('./SpeciesIcon.jsx');

//*------------------------------------------------------------------*

module.exports = function(mountNode,species,colourOverride, colourPerKingdom) {
    ReactDOM.render(
        React.createElement(Icon,{
          species: species,
          colourOverride: colourOverride,
          colourPerKingdom: colourPerKingdom
        }), mountNode
    );
};
