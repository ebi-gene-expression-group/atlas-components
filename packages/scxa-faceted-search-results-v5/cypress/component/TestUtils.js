import facets from '../fixtures/vindicatorsResponse.json'
import organismParts from '../fixtures/organismPartsResponse.json'
import cellTypes from '../fixtures/cellTypesResponse.json'
import species from '../fixtures/speciesResponse.json'

import PropTypes from "prop-types";
import styled from "styled-components";
import React from "react";
import CheckboxFacetGroupsDefaultProps from "../../src/facetgroups/CheckboxFacetGroupsDefaultProps";
import MultiSelectDropdownFacetGroupsDefaultProps
  from "../../src/facetgroups/MultiSelectDropdownFacetGroupsDefaultProps";

const getPropsForCheckBoxGroupWithTooltip = () => {
  return Cypress._.shuffle(CheckboxFacetGroupsDefaultProps).pop()
}

const getPropsForMultiSelectDropdownGroupWithTooltip = () => {
  return Cypress._.shuffle(MultiSelectDropdownFacetGroupsDefaultProps).pop()
}

const getPropsForMarkerGeneFacet = () => {
  const markerGeneName = `Marker genes`;
  return CheckboxFacetGroupsDefaultProps.find(facet => facet.name === markerGeneName)
}

const getPropsWithoutTooltip = () => {
  let props = getPropsForCheckBoxGroupWithTooltip()
  props.description = ``
  return props
}

const getFacets = () => {
  return Cypress._.shuffle(facets).filter(() => Math.random() > 0.5)
}

const getRandomSpecies = () => {
  return Cypress._.shuffle(species).filter(() => Math.random() > 0.5)
}

const getRandomOrganismParts = () => {
  return Cypress._.shuffle(organismParts).filter(() => Math.random() > 0.5)
}

const getRandomCellTypes = () => {
  return Cypress._.shuffle(cellTypes).filter(() => Math.random() > 0.5)
}

const ExperimentTableCard = ({title}) =>
  <div>
    <p>{title}</p>
  </div>

ExperimentTableCard.propTypes = {
  title: PropTypes.string.isRequired
}

const IconDiv = styled.div`
  width: 15%;
  text-align: center;
`
IconDiv.displayName = 'IconDiv'

const BitDiv = styled.div`
  width: 25%;
  text-align: center;
`
const DoveDiv = styled.div`
  width: 25%;
  text-align: center;
`

const ExperimentTableHeader = ({onClick}) =>
  <DoveDiv>
    <IconDiv onClick={()=>onClick(`test`)}>
      Meow
    </IconDiv>
    <BitDiv onClick={()=>onClick(`test`)}>
      Wow wow
    </BitDiv>
  </DoveDiv>

ExperimentTableHeader.propTypes = {
  onClick: PropTypes.func.isRequired
}

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

export { getRandomInt, getFacets, getRandomOrganismParts, getRandomCellTypes, getRandomSpecies,
  getPropsForCheckBoxGroupWithTooltip, getPropsForMultiSelectDropdownGroupWithTooltip,
  getPropsWithoutTooltip, getPropsForMarkerGeneFacet,
  ExperimentTableCard, ExperimentTableHeader }
