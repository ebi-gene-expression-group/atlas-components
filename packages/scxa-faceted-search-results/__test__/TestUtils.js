import React from 'react'
import PropTypes from 'prop-types'

const EpisodeCard = ({title}) =>
  <div>
    <p>{title}</p>
  </div>

EpisodeCard.propTypes = {
  title: PropTypes.string.isRequired
}

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min)) + min; //The maximum is exclusive and the minimum is inclusive
}

const vindicators = [
  { group:`Vindicator`, value: `supernova`, label: `Supernova`},
  { group:`Vindicator`, value: `noob-noob`, label: `Noob-Noob` },
  { group:`Vindicator`, value: `vance_maximus`, label: `Vance Maximus`},
  { group:`Vindicator`, value: `alan_rails`, label: `Alan Rails`},
  { group:`Vindicator`, value: `crocubot`, label: `Crocubot` },
  { group:`Vindicator`, value: `million_ants`, label: `Million Ants`},
  { group:`Vindicator`, value: `morty_smith`, label: `Morty Smith`},
  { group:`Vindicator`, value: `rick_sanchez`, label: `Rick Sanchez`},
  { group:`Vindicator`, value: `lady_katana`, label: `Lady Katana`},
  { group:`Vindicator`, value: `calypso`, label: `Calypso`},
  { group:`Vindicator`, value: `diablo_verde`, label: `Diablo Verde`}
]

const episodes = [
  {
    element: {
      title: "Raising Gazorpazorp"
    },
    facets: [
      {
        group: "Planet",
        value: "gazorpazorp",
        label: "Gazorpazorp"
      },
      {
        group: "Guest character",
        value: "gwendolyn",
        label: "Gwendolyn"
      },
      {
        group: "Guest character",
        value: "ma-sha",
        label: "Ma-Sha"
      },
      {
        group: "Season",
        value: "1",
        label: "1"
      }
    ]
  },
  {
    element: {
      title: "The wedding squanchers"
    },
    facets: [
      {
        group: "Planet",
        value: "squanch",
        label: "Squanch"
      },
      {
        group: "Guest character",
        value: "birdperson",
        label: "Birdperson"
      },
      {
        group: "Guest character",
        value: "squanchy",
        label: "Squanchy"
      },
      {
        group: "Season",
        value: "2",
        label: "2"
      }
    ]
  },
  {
    element: {
      title: "The Rickshank redemption"
    },
    facets: [
      {
        group: "Guest character",
        value: "birdperson",
        label: "Birdperson"
      },
      {
        group: "Planet",
        value: "buttworld",
        label: "Buttworld"
      },
      {
        group: "Season",
        value: "3",
        label: "3"
      }
    ]
  },
  {
    element: {
      title: "Ricksy business"
    },
    facets: [
      {
        group: "Guest character",
        value: "squanchy",
        label: "Squanchy"
      },
      {
        group: "Guest character",
        value: "abradolf_lincler",
        label: "Abradolf Lincler"
      },
      {
        group: "Season",
        value: "1",
        label: "1"
      }
    ]
  },
  {
    element: {
      title: "Close Rick-counters of the Rick kind"
    },
    facets: [
      {
        group: "Guest character",
        value: "ricktiminus_sancheziminius",
        label: "Ricktiminus Sancheziminius"
      },
      {
        group: "Guest character",
        value: "abradolf_lincler",
        label: "Abradolf Lincler"
      },
      {
        group: "Planet",
        value: "buttworld",
        label: "Buttworld"
      },
      {
        group: "Season",
        value: "1",
        label: "1"
      }
    ]
  }
]



export {getRandomInt, vindicators, episodes, EpisodeCard}
