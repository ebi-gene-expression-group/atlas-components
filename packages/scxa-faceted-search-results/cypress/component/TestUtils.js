import episodes from '../fixtures/episodes.json'
import PropTypes from "prop-types";
import styled from "styled-components";
import React from "react";

const facets = () => {
  return episodes.reduce((acc, episode) => acc.concat(episode.facets), [])
}

const getUniqueFacets = () => {
  const allFacets = facets()
  return allFacets
    .filter((facet, index) => allFacets.findIndex((thatFacet) => facet.value === thatFacet.value) === index)
    .map((facet) => ({
      ...facet,
      disabled: false
    }))
}

const getFacetTooltip = () => {
  return Cypress._.shuffle(getUniqueFacets()).find(facet => facet.description)
}

const getFacetWithoutTooltip = () => {
  return Cypress._.shuffle(getUniqueFacets()).find(facet => !facet.description)
}

const getPropsWithTooltip = (facetTooltip) => {
  return {
    facetGroupName: facetTooltip.group,
      facetGroupNameDescription: facetTooltip.description,
    facets: getUniqueFacets().filter(facet => facet.group === facetTooltip.group),
    onChange: () => {}
  }
}

const getPropsWithoutTooltip = () => {
  const facetWithoutTooltip = getFacetWithoutTooltip()
  return {
    facetGroupName: facetWithoutTooltip.group,
      facets: getUniqueFacets().filter(facet => facet.group === facetWithoutTooltip.group),
    onChange: () => {}
  }
}

const ExperimentTableCard = ({title}) =>
  <div>
    <p>{title}</p>
  </div>

ExperimentTableCard.propTypes = {
  title: PropTypes.string.isRequired
}

const IconDiv = styled.div`
  width: 15%;
  text-align: center;
`
IconDiv.displayName = 'IconDiv'

const BitDiv = styled.div`
  width: 25%;
  text-align: center;
`
const DoveDiv = styled.div`
  width: 25%;
  text-align: center;
`

const ExperimentTableHeader = ({onClick}) =>
  <DoveDiv>
    <IconDiv onClick={()=>onClick(`test`)}>
      Meow
    </IconDiv>
    <BitDiv onClick={()=>onClick(`test`)}>
      Wow wow
    </BitDiv>
  </DoveDiv>

ExperimentTableHeader.propTypes = {
  onClick: PropTypes.func.isRequired
}

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

export { episodes, getRandomInt, getUniqueFacets, getFacetTooltip, getPropsWithTooltip, getPropsWithoutTooltip,
  ExperimentTableCard, ExperimentTableHeader }
