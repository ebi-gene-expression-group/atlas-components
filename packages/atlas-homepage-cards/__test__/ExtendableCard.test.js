import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import EbiSpeciesIcon from 'react-ebi-species'

import { getRandomInt } from './TestUtils'
import {
  aRickleInTimeImageCardProps, theSmithHouseholdImageCardProps,
  batmanFilmsSpeciesCardProps, findingNemoSpeciesCardProps
 } from './TestUtils'
import ExtendableCard from '../src/cards/ExtendableCard'

Enzyme.configure({ adapter: new Adapter() })

describe(`ExtendableCard`, () => {
  test.each([ [aRickleInTimeImageCardProps.description.text, aRickleInTimeImageCardProps],
              [theSmithHouseholdImageCardProps.description.text, theSmithHouseholdImageCardProps],
              [batmanFilmsSpeciesCardProps.description.text, batmanFilmsSpeciesCardProps],
              [findingNemoSpeciesCardProps.description.text, findingNemoSpeciesCardProps] ])(
    `matches snapshot: %s`, (titleText, props) => {
    const tree = renderer.create(<ExtendableCard {...props}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test(`can render species cards`, () => {
    const props = [batmanFilmsSpeciesCardProps, findingNemoSpeciesCardProps][getRandomInt(0, 2)]
    const wrapper = shallow(<ExtendableCard {...props}/>)
    expect(wrapper.find(EbiSpeciesIcon).exists()).toBe(true)
    expect(wrapper.find(`img`).exists()).toBe(false)
  })

  test(`can render image cards`, () => {
    const props = [aRickleInTimeImageCardProps, theSmithHouseholdImageCardProps][getRandomInt(0, 2)]
    const wrapper = shallow(<ExtendableCard {...props}/>)
    expect(wrapper.find(EbiSpeciesIcon).exists()).toBe(false)
    expect(wrapper.find(`img`).exists()).toBe(true)

  })

  test(`does not render optional empty title and content`, () => {
    const props = {
      iconType: [`species`, `image`][getRandomInt(0, 2)],
      iconSrc: ``
    }
    const wrapper = shallow(<ExtendableCard {...props} />)

    expect(wrapper.find(`h4`).exists()).toBe(false)
    expect(wrapper.find(`li`).exists()).toBe(false)
  })

  test(`does not render non-existent URLs in title`, () => {
    const props = [theSmithHouseholdImageCardProps, batmanFilmsSpeciesCardProps][getRandomInt(0, 2)]

    const wrapper = shallow(<ExtendableCard {...props} />)

    expect(wrapper.find(`h4`).exists()).toBe(true)
    expect(wrapper.find(`h4`).text()).toBe(props.description.text)
    expect(wrapper.find(`h4 a`).exists()).toBe(false)
  })

  test(`does not render non-existent URLs in content`, () => {
    const props = [aRickleInTimeImageCardProps, findingNemoSpeciesCardProps][getRandomInt(0, 2)]

    const wrapper = shallow(<ExtendableCard {...props} />).find(`ul`)

    expect(wrapper.exists()).toBe(true)
    expect(wrapper.find(`li`)).toHaveLength(Math.min(props.content.length, 5))
    expect(wrapper.find(`li a`).exists()).toBe(false)
  })

  test(`displays up to five content items by default with a button to show/hide the others`, () => {
    const props = [aRickleInTimeImageCardProps, batmanFilmsSpeciesCardProps][getRandomInt(0, 2)]
    expect(props.content.length).toBeGreaterThan(5)

    const wrapper = shallow(<ExtendableCard {...props} />)
    expect(wrapper.find(`li`)).toHaveLength(5)
    wrapper.find(`button`).at(0).simulate(`click`)
    wrapper.update()
    expect(wrapper.find(`li`)).toHaveLength(props.content.length)
    wrapper.find(`button`).at(0).simulate(`click`)
    wrapper.update()
    expect(wrapper.find(`li`)).toHaveLength(5)
  })
})
