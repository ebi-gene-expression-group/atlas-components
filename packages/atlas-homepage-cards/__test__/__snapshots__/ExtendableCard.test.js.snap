// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtendableCard matches snapshot: A rickle in time 1`] = `
<div
  style={
    Object {
      "marginBottom": 0,
      "paddingBottom": "2rem",
      "textAlign": "center",
    }
  }
>
  <h4>
    <a
      href="http://rickandmorty.wikia.com/wiki/A_Rickle_in_Time"
    >
      A rickle in time
    </a>
  </h4>
  <a
    href="http://rickandmorty.wikia.com/wiki/A_Rickle_in_Time"
    style={
      Object {
        "borderBottom": 0,
      }
    }
  >
    <img
      src="https://upload.wikimedia.org/wikipedia/en/6/6b/A_Rickle_in_Time.jpg"
      style={
        Object {
          "height": "8rem",
          "marginBottom": "2.35rem",
          "marginTop": "2.50rem",
        }
      }
    />
  </a>
  <ul
    className="content sc-htpNat dUITwr"
  >
    <li
      style={
        Object {
          "marginBottom": "0.3rem",
        }
      }
    >
      Morty Smith
    </li>
    <li
      style={
        Object {
          "marginBottom": "0.3rem",
        }
      }
    >
      Summer Smith
    </li>
    <li
      style={
        Object {
          "marginBottom": "0.3rem",
        }
      }
    >
      Rick Sanchez
    </li>
    <li
      style={
        Object {
          "marginBottom": "0.3rem",
        }
      }
    >
      Rick Sanchez(split reality)
    </li>
    <li
      style={
        Object {
          "marginBottom": "0.3rem",
        }
      }
    >
      Morty Smith (split reality)
    </li>
  </ul>
  <button
    className="button"
    onClick={[Function]}
  >
    Show all
  </button>
</div>
`;

exports[`ExtendableCard matches snapshot: Batman films 1`] = `
<div
  style={
    Object {
      "marginBottom": 0,
      "paddingBottom": "2rem",
      "textAlign": "center",
    }
  }
>
  <h4>
    Batman films
  </h4>
  <span
    style={
      Object {
        "fontSize": "8rem",
      }
    }
  >
    <span
      className="icon icon-species"
      data-icon="("
      style={
        Object {
          "color": "indianred",
        }
      }
      title="Bat"
    />
  </span>
  <ul
    className="content sc-htpNat dUITwr"
  >
    <li
      style={
        Object {
          "marginBottom": "0.3rem",
        }
      }
    >
      <a
        href="https://en.wikipedia.org/wiki/Batman_(serial)"
      >
        Batman (1943)
      </a>
    </li>
    <li
      style={
        Object {
          "marginBottom": "0.3rem",
        }
      }
    >
      <a
        href="https://en.wikipedia.org/wiki/Batman_and_Robin_(serial)"
      >
        Batman and Robin (1949)
      </a>
    </li>
    <li
      style={
        Object {
          "marginBottom": "0.3rem",
        }
      }
    >
      <a
        href="https://en.wikipedia.org/wiki/Batman_(1966_film)"
      >
        Batman (1966)
      </a>
    </li>
    <li
      style={
        Object {
          "marginBottom": "0.3rem",
        }
      }
    >
      <a
        href="https://en.wikipedia.org/wiki/Batman_(1989_film)"
      >
        Batman (1989)
      </a>
    </li>
    <li
      style={
        Object {
          "marginBottom": "0.3rem",
        }
      }
    >
      <a
        href="https://en.wikipedia.org/wiki/Batman_Returns"
      >
        Batman Returns (1992)
      </a>
    </li>
  </ul>
  <button
    className="button"
    onClick={[Function]}
  >
    Show all
  </button>
</div>
`;

exports[`ExtendableCard matches snapshot: Finding Nemo 1`] = `
<div
  style={
    Object {
      "marginBottom": 0,
      "paddingBottom": "2rem",
      "textAlign": "center",
    }
  }
>
  <h4>
    <a
      href="https://en.wikipedia.org/wiki/Finding_Nemo"
    >
      Finding Nemo
    </a>
  </h4>
  <a
    href="https://en.wikipedia.org/wiki/Finding_Nemo"
    style={
      Object {
        "borderBottom": 0,
      }
    }
  >
    <span
      style={
        Object {
          "fontSize": "8rem",
        }
      }
    >
      <span
        className="icon icon-species"
        data-icon="Z"
        style={
          Object {
            "color": "deepskyblue",
          }
        }
        title="Zebrafish"
      />
    </span>
  </a>
  <ul
    className="content sc-htpNat dUITwr"
  >
    <li
      style={
        Object {
          "marginBottom": "0.3rem",
        }
      }
    >
      Dory
    </li>
    <li
      style={
        Object {
          "marginBottom": "0.3rem",
        }
      }
    >
      Nemo
    </li>
    <li
      style={
        Object {
          "marginBottom": "0.3rem",
        }
      }
    >
      Gill
    </li>
  </ul>
</div>
`;

exports[`ExtendableCard matches snapshot: The Smith household 1`] = `
<div
  style={
    Object {
      "marginBottom": 0,
      "paddingBottom": "2rem",
      "textAlign": "center",
    }
  }
>
  <h4>
    The Smith household
  </h4>
  <img
    src="https://upload.wikimedia.org/wikipedia/en/b/b0/Rick_and_Morty_characters.jpg"
    style={
      Object {
        "height": "8rem",
        "marginBottom": "2.35rem",
        "marginTop": "2.50rem",
      }
    }
  />
  <ul
    className="content sc-htpNat dUITwr"
  >
    <li
      style={
        Object {
          "marginBottom": "0.3rem",
        }
      }
    >
      <a
        href="https://en.wikipedia.org/wiki/List_of_Rick_and_Morty_characters#Jerry_Smith"
      >
        Jerry Smith
      </a>
    </li>
    <li
      style={
        Object {
          "marginBottom": "0.3rem",
        }
      }
    >
      <a
        href="https://en.wikipedia.org/wiki/List_of_Rick_and_Morty_characters#Beth_Smith"
      >
        Beth Smith
      </a>
    </li>
    <li
      style={
        Object {
          "marginBottom": "0.3rem",
        }
      }
    >
      <a
        href="https://en.wikipedia.org/wiki/List_of_Rick_and_Morty_characters#Summer_Smith"
      >
        Summer Smith
      </a>
    </li>
    <li
      style={
        Object {
          "marginBottom": "0.3rem",
        }
      }
    >
      <a
        href="https://en.wikipedia.org/wiki/Morty_Smith"
      >
        Morty Smith
      </a>
    </li>
    <li
      style={
        Object {
          "marginBottom": "0.3rem",
        }
      }
    >
      <a
        href="https://en.wikipedia.org/wiki/Rick_Sanchez_(Rick_and_Morty)"
      >
        Rick Sanchez
      </a>
    </li>
  </ul>
</div>
`;
